src/problem_bank_helpers/problem_bank_helpers.py

```diff
@@ -51,11 +52,12 @@ def sigfigs(x):
     if ('e' in x):
         # return the length of the numbers before the 'e'
         myStr = x.split('e')
-        return len( myStr[0] ) - 1 # to compenstate for the decimal point
+        
+        return len( myStr[0] ) - (1 if '.' in x else 0) # to compenstate for the decimal point
     else:
         # put it in e format and return the result of that
-        ### NOTE: because of the 8 below, it may do crazy things when it parses 9 sigfigs
-        n = ('%.*e' %(8, float(x))).split('e')
+        ### NOTE: because of the 15 below, it may do crazy things when it parses 16 sigfigs
+        n = ('%.*e' %(15, float(x))).split('e')
```

Comment:
```suggestion
        n = f'{float(x):.15e}'.split('e')
```
Seems more readable like this:
```python
>>> x = '1.234567890'
>>> ('%.*e' %(15, float(x))).split('e')
['1.234567890000000', '+00']
>>> f'{float(x):.15e}'.split('e')
['1.234567890000000', '+00']
```